tikv/gc.go

```diff
@@ -94,72 +95,115 @@ func (s *KVStore) resolveLocks(ctx context.Context, safePoint uint64, concurrenc
 	return nil
 }
 
+type BaseLockResolver struct {
+	store Storage
+	bo    *Backoffer
+}
+
+func NewBaseLockResolver(store Storage) *BaseLockResolver {
+	return &BaseLockResolver{
+		store: store,
+	}
+}
+
+func (l *BaseLockResolver) ResolveLocks(ctx context.Context, locks []*txnlock.Lock, loc *locate.KeyLocation) (*locate.KeyLocation, error) {
+	// if ScanLocks created backoffer before, reuse it
+	// otherwise create it to prevent panic.
+	if l.bo == nil {
+		l.bo = NewGcResolveLockMaxBackoffer(ctx)
+	}
+	return batchResolveLocksInARegion(l.bo, l.GetStore(), locks, loc)
+}
+
+func (l *BaseLockResolver) ScanLocks(ctx context.Context, key []byte, maxVersion uint64) ([]*txnlock.Lock, *locate.KeyLocation, error) {
+	// every time ScanLocks need a new backoffer
+	l.bo = NewGcResolveLockMaxBackoffer(ctx)
```

Comment:
How about providing the backoff from the caller?
I'm afraid it might easily cause misusing to making it a state of the lock resolver.

Reply:
the origin logic is `ScanLocks` and `ResolveLocks` **must use backoff and must share one backoff**, and gc worker test using failpoint to control the backoff. 
So I think backoff is an internal logic for GCWorker and the test doesn't need it.

If we public it, then it should be caller's responsibility to call `SetBackoffer` and `GetBackoffer` when using `ScanLocks` and `ResolveLocks` for LockResolver? and caller may not know they need **share** the same backoff. Do you think it's still ok?